src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1195,6 +1223,107 @@ def construct_payload(self):
+            return None
+
+    def _post_process_for_repo(self):
+        from ._up_utils import (get_token, _create_github_action)
+        app, env = self._construct_app_and_env_for_source_or_repo()
+        # Get GitHub access token
+        token = get_token(self.cmd, self.get_argument_repo(), self.get_argument_token())
+        _create_github_action(app, env, self.get_argument_service_principal_client_id(), self.get_argument_service_principal_client_secret(),
+                              self.get_argument_service_principal_tenant_id(), self.get_argument_branch(), token, self.get_argument_repo(), self.get_argument_context_path())
+        cache_github_token(self.cmd, token, self.get_argument_repo())
+        return self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+
+    def _construct_app_and_env_for_source_or_repo(self):
+        from ._up_utils import (ContainerApp, ResourceGroup, ContainerAppEnvironment, _reformat_image, get_token, _has_dockerfile, _get_dockerfile_content, _get_ingress_and_target_port, _get_registry_details, _create_github_action)
+        ingress = self.get_argument_ingress()
+        target_port = self.get_argument_target_port()
+        dockerfile = "Dockerfile"
+        token = get_token(self.cmd, self.get_argument_repo(), self.get_argument_token())
+
+        # Parse resource group name and managed env name
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Parse location
+        env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+        location = self.containerapp_def["location"] if "location" in self.containerapp_def else env_info['location']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
+
+        has_dockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
+        if not self.get_argument_source() or has_dockerfile:
+            dockerfile_content = _get_dockerfile_content(self.get_argument_repo(), self.get_argument_branch(), token, self.get_argument_source(), self.get_argument_context_path(), dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
+
+        # Construct ContainerApp
+        resource_group = ResourceGroup(self.cmd, self.get_argument_resource_group_name(), location=location)
+        env = ContainerAppEnvironment(self.cmd, env_name, resource_group, location=location)
```

Comment:
```suggestion
        env_resource_group = ResourceGroup(self.cmd, env_rg, location=location)
        env = ContainerAppEnvironment(self.cmd, env_name, env_resource_group , location=location)
```

Reply:
Please use 2 different values for env's and app's resource_group .

Reply:
app's resource group should not be `self.get_argument_resource_group_name()` because `ContainerApp` expects [ResourceGroup](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/_up_utils.py#L352). This is how it has been implemented for `up` as well [here](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/custom.py#L3536).